src/ripple/app/paths/PathRequest.cpp

```diff
@@ -474,25 +474,30 @@ PathRequest::getPathFinder(RippleLineCache::ref cache,
     return currency_map[currency] = std::move(pathfinder);
 }
 
-void
+bool
 PathRequest::findPaths (RippleLineCache::ref cache, int const level,
     Json::Value& jvArray)
 {
-    auto sourceCurrencies = sciSourceCurrencies;
-    if (sourceCurrencies.empty ())
+    std::set<ripple::Issue> sourceCurrencies;
+    if (sciSourceCurrencies.empty ())
     {
-        auto usCurrencies =
-            accountSourceCurrencies(*raSrcAccount, cache, true);
-        bool sameAccount = *raSrcAccount == *raDstAccount;
+        auto usCurrencies = accountSourceCurrencies(*raSrcAccount, cache, true);
+        bool const sameAccount = *raSrcAccount == *raDstAccount;
         for (auto const& c : usCurrencies)
         {
-            if (!sameAccount || (c != saDstAmount.getCurrency()))
+            if (! sameAccount || c != saDstAmount.getCurrency())
             {
+                if (sourceCurrencies.size() >= RPC::Tuning::max_auto_src_cur)
+                    return false;
```

Comment:
No, the number of source currencies is comprised from lines with unique currencies and a positive balance or credit. So for instance you can have 1000 USD lines with a positive balance or credit and that would only count as one source currency.

David did some analytical work examining the actual number of trust lines accounts have. The highest was 88 and there was only one. Of those 88 lines, I found only forty something had a positive balance or credit. The next account with the highest had less than half of that. The average is about 18. I consider 88 absurdly high.

Once I implement proper resource fee for work performed, this number may need to get tweaked as having that many source currencies would probably drop the client anyway.